server/src/main/java/org/opensearch/index/fielddata/fieldcomparator/DoubleValuesComparatorSource.java

```diff
@@ -104,7 +104,7 @@ public FieldComparator<?> newComparator(String fieldname, int numHits, boolean e
         final double dMissingValue = (Double) missingObject(missingValue, reversed);
         // NOTE: it's important to pass null as a missing value in the constructor so that
         // the comparator doesn't check docsWithField since we replace missing values in select()
-        return new DoubleComparator(numHits, fieldname, null, reversed, enableSkipping && this.enableSkipping) {
+        return new DoubleComparator(numHits, fieldname, dMissingValue, reversed, enableSkipping && this.enableSkipping) {
```

Comment:
is this change intended? If yes, can you add tests for the same?

Reply:
Yeah this is intended, adding tests.

Reply:
I added big test for this, https://github.com/opensearch-project/OpenSearch/pull/11024/commits/a2015ef1c630b169a711e5487312acd3363f608a
For all numeric types. 
Basically, passing missing value is requred in case we have missing values determined, which is the case here.
If we dont pass missing value here, it might hit the corner case like here https://github.com/opensearch-project/OpenSearch/issues/9537. We fixed this in Lucene, https://github.com/apache/lucene/pull/12520, but didnt add integ tests here in OpenSearch. 

Thanks @backslasht for reminding about adding integ tests here. This is definitely one of misssing corner case but important one.